### PR TITLE
Added message saying that link for password restoring is no longer ac…

### PR DESCRIPTION
**Pull request by Misha999777**
…tive

### CODE REVIEW

File: src/main/java/com/gmail/buer2012/controller/AuthController.java [hunk 1]
```diff
@@ -148,10 +148,17 @@ private void sendConfirmationEmail(String requestEmail, User user) {
     @PostMapping("/confirmRestore")
     public ResponseEntity<?> restorePassword(@RequestBody RestorePasswordRequest restorePasswordRequest) {
         EmailConfirmationToken emailConfirmationToken = emailConfirmationTokenRepository.findByToken(restorePasswordRequest.getToken());
```
Comment (BukerKo) [inline src/main/java/com/gmail/buer2012/controller/AuthController.java]: Probably make findByToken returning Optional and check whether it is present?

File: src/main/java/com/gmail/buer2012/controller/AuthController.java [hunk 2]
```diff
@@ -148,10 +148,17 @@ private void sendConfirmationEmail(String requestEmail, User user) {
...
-        user.setPassword(passwordEncoder.encode(restorePasswordRequest.getPassword()));
-        userService.updateUser(user);
-        return ResponseEntity.ok(0);
+        if(emailConfirmationToken != null) {
+            User user = emailConfirmationToken.getUser();
+            emailConfirmationTokenRepository.delete(emailConfirmationToken);
+            user.setPassword(
+                passwordEncoder.encode(restorePasswordRequest.getPassword()));
+            userService.updateUser(user);
+            return ResponseEntity.ok(0);
```
Comment (BukerKo) [inline src/main/java/com/gmail/buer2012/controller/AuthController.java:168]: Why always using 0? Can't it be left blank?

Comment (Misha999777) [reply src/main/java/com/gmail/buer2012/controller/AuthController.java:168]: It must have some body

Comment (BukerKo) [reply src/main/java/com/gmail/buer2012/controller/AuthController.java:168]: `return ResponseEntity.ok().build();`

**Review by BukerKo — APPROVED** [review]

**Pull request merged**